--- AnkiDroid/src/test/java/com/ichi2/anki/deckpicker/DeckPickerViewModelTest.kt
+++ AnkiDroid/src/test/java/com/ichi2/anki/deckpicker/DeckPickerViewModelTest.kt
@@ -114,6 +114,7 @@
             viewModel.emptyFilteredDeck(filteredDeckId)
             advanceUntilIdle()

             assertThat("deck was reset", note.firstCard().did, equalTo(Consts.DEFAULT_DECK_ID))
+            viewModel.loadDeckCounts?.cancel()
         }
     }

@@ -134,6 +135,7 @@
             val updatedDueTree = viewModel.dueTree
             val updatedFilteredDeckNode = updatedDueTree?.find(filteredDeckId)
             assertThat("filtered deck has no cards after", updatedFilteredDeckNode?.hasCardsReadyToStudy(), equalTo(false))
+            viewModel.loadDeckCounts?.cancel()
         }
     }

@@ -145,6 +147,7 @@
             ensureOpsExecuted(1) {
                 viewModel.emptyFilteredDeck(filteredDeckId)
                 advanceUntilIdle()
             }

             // backend assert
             assertThat("col undo status", col.undoStatus().undo, equalTo("Empty"))
+            viewModel.loadDeckCounts?.cancel()
         }
     }
